src/session/content/message_row/document.rs

```diff
@@ -0,0 +1,355 @@
+use glib::clone;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gio, glib, CompositeTemplate};
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use tdlib::enums::MessageContent;
+use tdlib::types::File;
+
+use crate::session::content::message_row::{
+    MessageBase, MessageBaseImpl, MessageIndicators, MessageLabel,
+};
+use crate::tdlib::{ChatType, Message, MessageSender};
+use crate::utils::parse_formatted_text;
+use crate::Session;
+
+use super::base::MessageBaseExt;
+
+mod imp {
+    use super::*;
+    use once_cell::sync::Lazy;
+    use once_cell::unsync::OnceCell;
+    use std::cell::RefCell;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(resource = "/com/github/melix99/telegrand/ui/content-message-document.ui")]
+    pub(crate) struct MessageDocument {
+        pub(super) sender_color_class: RefCell<Option<String>>,
+        pub(super) bindings: RefCell<Vec<gtk::ExpressionWatch>>,
+        pub(super) status_handler_id: RefCell<Option<glib::SignalHandlerId>>,
+        pub(super) click_handler_id: OnceCell<glib::SignalHandlerId>,
+        pub(super) message: RefCell<Option<glib::Object>>,
+        #[template_child]
+        pub(super) sender_label: TemplateChild<gtk::Label>,
+        #[template_child]
+        pub(super) content_label: TemplateChild<MessageLabel>,
+        #[template_child]
+        pub(super) indicators: TemplateChild<MessageIndicators>,
+        #[template_child]
+        pub(super) file_name_label: TemplateChild<gtk::Label>,
+        #[template_child]
+        pub(super) file_size_label: TemplateChild<gtk::Label>,
+        #[template_child]
+        pub(super) file_status_button: TemplateChild<gtk::Button>,
+        #[template_child]
+        pub(super) document_box: TemplateChild<gtk::Box>,
+        #[template_child]
+        pub(super) click: TemplateChild<gtk::GestureClick>,
```

Comment:
The fields in the struct should match the order in the ui file.
I think, this is the last chance request for me that I think it looks good.
I think it's a good time to start cleaning (squash, fixup, etc.) up the commit you made.
 
```suggestion
        #[template_child]
        pub(super) sender_label: TemplateChild<gtk::Label>,
        #[template_child]
        pub(super) document_box: TemplateChild<gtk::Box>,
        #[template_child]
        pub(super) click: TemplateChild<gtk::GestureClick>,
        #[template_child]
        pub(super) file_status_button: TemplateChild<gtk::Button>,
        #[template_child]
        pub(super) file_name_label: TemplateChild<gtk::Label>,
        #[template_child]
        pub(super) file_size_label: TemplateChild<gtk::Label>,
        #[template_child]
        pub(super) content_label: TemplateChild<MessageLabel>,
        #[template_child]
        pub(super) indicators: TemplateChild<MessageIndicators>,
```